### PR TITLE
M3-5367 E2E image upload with post processing event

### PR DESCRIPTION
**Pull request by ndukatz**
## Description
This uploads a real image to kick off each test, and then mocks the event after landing on the /images page. There are 4 tests: 3 failed uploads and 1 success. The first test upload succeeds, but I needed to also mock an image for that one on the /images page. I'm not  100% sure if this is properly simulating the actual flow for these cases so please take a look at that when you review.

**What does this PR do?**
Tests to test image upload. 

## How to test
- `yarn up` in one terminal
- `yarn cy:e2e -s ./cypress/integration/images/machine-image-upload.spec.ts` in another
- Result should be "✔  All specs passed!"

## Make sure your .env contains:
```
REACT_APP_LOGIN_ROOT
REACT_APP_API_ROOT
REACT_APP_APP_ROOT
REACT_APP_LAUNCH_DARKLY_ID
REACT_APP_CLIENT_ID
MANAGER_OAUTH
```

### CODE REVIEW

File: packages/manager/cypress/integration/images/machine-image-upload.spec.ts [hunk 1]
```diff
@@ -1,65 +1,156 @@
...
-    // }).as('mockImage');
+const interceptOnce = (method, url, response) => {
+  let count = 0;
+  return cy.intercept(method, url, (req) => {
+    count += 1;
+    if (count < 2) {
+      req.reply(response);
+    }
+  });
+};
```
Comment (bnussman) [inline packages/manager/cypress/integration/images/machine-image-upload.spec.ts:21]: I think I've seen this used in other places. We should move it somewhere for helper functions and import it where needed.

File: packages/manager/cypress/integration/images/machine-image-upload.spec.ts [hunk 2]
```diff
@@ -1,65 +1,156 @@
...
+
+const eventIntercept = (id, status, message?, created?) => {
+  interceptOnce(
```
Comment (bnussman) [inline packages/manager/cypress/integration/images/machine-image-upload.spec.ts:24]: If it wouldn't be too much trouble, add Typescript types to the parameters of functions.

**Review by bnussman — APPROVED** [review]
Works nicely!

Should we include a test that does not much the upload success or error so the api gets fully used?

I also can't tell why the success / failure toast happens nearly instantly but the table takes a while to get updated... Can you explain why? I though the toast and table are both updated based on the same events request.

**Review by DevDW — COMMENTED** [review]
When I ran the test, I got a series of failures:

```
  1) machine image
       uploads machine image, mock finished event:
     CypressError: Timed out retrying after 10000ms: `cy.wait()` timed out waiting `10000ms` for the 1st request to the route: `getImage`. No request ever occurred.

https://on.cypress.io/wait
      at cypressErr (http://localhost:3000/__cypress/runner/cypress_runner.js:160503:18)
      at Module.errByPath (http://localhost:3000/__cypress/runner/cypress_runner.js:160572:10)
      at checkForXhr (http://localhost:3000/__cypress/runner/cypress_runner.js:147088:33)
      at http://localhost:3000/__cypress/runner/cypress_runner.js:147111:28
      at tryCatcher (http://localhost:3000/__cypress/runner/cypress_runner.js:13212:23)
      at Function.Promise.attempt.Promise.try (http://localhost:3000/__cypress/runner/cypress_runner.js:10486:29)
      at tryFn (http://localhost:3000/__cypress/runner/cypress_runner.js:153198:21)
      at whenStable (http://localhost:3000/__cypress/runner/cypress_runner.js:153236:12)
      at http://localhost:3000/__cypress/runner/cypress_runner.js:152728:16
      at tryCatcher (http://localhost:3000/__cypress/runner/cypress_runner.js:13212:23)
      at Promise._settlePromiseFromHandler (http://localhost:3000/__cypress/runner/cypress_runner.js:11147:31)
      at Promise._settlePromise (http://localhost:3000/__cypress/runner/cypress_runner.js:11204:18)
      at Promise._settlePromise0 (http://localhost:3000/__cypress/runner/cypress_runner.js:11249:10)
      at Promise._settlePromises (http://localhost:3000/__cypress/runner/cypress_runner.js:11329:18)
      at Promise._fulfill (http://localhost:3000/__cypress/runner/cypress_runner.js:11273:18)
      at http://localhost:3000/__cypress/runner/cypress_runner.js:12887:46
  From Your Spec Code:
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18863:16)

  2) machine image
       uploads machine image, mock upload cancelled failed event:
     AssertionError: Timed out retrying after 80000ms: Expected to find element: `[data-qa-image-cell="private/13684633"]`, but never found it.
      at assertProcessing (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18828:8)
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18880:13)

  3) machine image
       uploads machine image, mock failed to decompress failed event:
     AssertionError: Timed out retrying after 80000ms: Expected to find element: `[data-qa-image-cell="private/13684645"]`, but never found it.
      at assertProcessing (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18828:8)
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18893:13)

  4) machine image
       uploads machine image, mock expired upload event:
     AssertionError: Timed out retrying after 80000ms: Expected to find element: `[data-qa-image-cell="private/13684651"]`, but never found it.
      at assertProcessing (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18828:8)
      at Context.eval (http://localhost:3000/__cypress/tests?p=cypress/integration/images/machine-image-upload.spec.ts:18907:13)
```

**Comment by ndukatz** [comment]
This should be working now. I had just forgotten to push some changes up that I had locally

File: packages/manager/cypress/integration/images/machine-image-upload.spec.ts [hunk 3]
```diff
@@ -1,65 +1,155 @@
-import { createMockImage } from '../../support/api/images';
+/* eslint-disable sonarjs/no-duplicate-string */
```
Comment (johnwcallahan) [inline packages/manager/cypress/integration/images/machine-image-upload.spec.ts:1]: I'm really starting to wonder if we should disable this rule everywhere.

**Review by johnwcallahan — APPROVED** [review]
Nice work. Tests passing, code looks good, and we're testing the right things. 

Like @bnussman I am wondering why a few of the assertions are taking a long time even though the conditions are matching, but I don't think that's blocking.

**Pull request merged**